docs: update contribution guide (#213)
lidofinance/core: CONTRIBUTING.md

### Opening an Issue

> [!CAUTION]
> For bugs in contracts and protocol vulnerabilities, please refer to the [Bug Bounty](/README.md#bug-bounty) program!

Issues are a great way to contribute to the project by reporting bugs or suggesting enhancements.

- **Bug Reports**: If you encounter a bug, please report it using the GitHub issues feature. Before submitting, check if
  the bug has already been reported. If it has, you can contribute by adding more details to the existing report. This
  only applies to off-chain code (tests, scripts, etc.). For on-chain code, please refer to
  the [Bug Bounty](/README.md#bug-bounty) program.
- **Feature Requests**: Have an idea for a new feature or an improvement to an existing one? Submit a feature request

>   through GitHub issues, detailing your proposed enhancements and how they would benefit the Lido Finance Core.

Reviewer: ```suggestion
  through GitHub issues, detailing your proposed enhancements and how they would benefit the Lido protocol middleware.
```

### Improving Documentation

Good documentation is crucial for any project. If you notice areas where the documentation can be improved, whether it's
fixing typos, adding examples, or clarifying explanations, your contributions are highly valued. These improvements help
make the project more accessible and understandable for everyone.

> For more extensive documentation, visit the [Lido Docs repository](https://github.com/lidofinance/docs).

Reviewer: ```suggestion
For more extensive documentation, visit the [Lido Docs repository](https://github.com/lidofinance/docs) deployed at [docs.lido.fi](https://docs.lido.fi).
```

### Contributing to the Codebase

Resolving open issues is a valuable way to contribute to the project. Browse through existing issues to find something
that interests you or matches your expertise. Don't hesitate to ask for more information or clarification before you
start working on an issue.

If you're interested in improving tooling and CI in this repository, consider opening a feature request issue first to
discuss your ideas with the community of contributors.

For larger ideas on how to improve the protocol, consider publishing your proposal in

> the [Lido Forum](https://research.lido.fi/).

Reviewer: ```suggestion
the [Lido Research Forum](https://research.lido.fi/).
```

- Resolve any issues flagged by the pre-commit hook before submitting a pull request

### Documentation

- Keep inline comments and documentation up-to-date with code changes
- For significant changes, update relevant sections of the project documentation

### Testing

- Write appropriate tests (unit tests, integration tests, and fuzzing tests) for new features or bug fixes
- Ensure all tests pass before submitting a pull request

Distinguishing Hardhat test files from Foundry-related files is crucial for proper test execution and maintaining code
consistency. For any questions, contact the maintainers.

## Branches

### `master`

This is the production branch and the default branch of the repository.
The [deployed protocol contracts](https://docs.lido.fi/deployed-contracts/) must match the contracts stored in the
`/contracts` directory. All pull requests to `master` must originate from the `develop` branch and require at least one
approving review before merging.

### `develop`

This is the main development branch. All pull requests to `master` should be submitted to `develop` first for peer
review.

## Local Deployment

> WIP

Reviewer: Why WIP? hadn't we had a scratch deploy?

Author: Yep, I forgot to add a link. Nice catch!

Author: Fixed in e20a363
